crowbar_framework/lib/crowbar/upgrade.rb

```diff
@@ -115,8 +115,20 @@ def upgrade_crowbar_files_1_9
       FileUtils.touch(@data.join("crowbar", "database.yml"))
     end
 
-    def create_network_json(json_path)
-      json = JSON.load(json_path.read)
+    def create_network_json(json_path, options = {})
+      from_role = options.fetch(:from_role, false)
+      json = if from_role
+        # get network proposal data from database.yml
+        proposals = YAML.load_file(
+          @data.join("crowbar", "database.yml")
+        )
+        network_proposal = proposals["proposals"]["records"].detect do |p|
+          p[1] == "network" && p[2] == "default"
```

Comment:
It is always the same order, like defined in the `schema.rb` database schema. and I just looked at 2 different database.yml files. so I guess it is not necessary to map it to the columns.